lib/openhab/dsl/monkey_patch/items/metadata.rb

```diff
@@ -134,6 +134,20 @@ def []=(namespace, value)
               end
             end
 
+            #
+            # Implements Hash#dig-like functionaity to metadata
+            #
+            # @param [String, Symbol] key The first key
+            # @param [Array<String, Symbol>] keys More keys to dig deeper
+            #
+            # @return [OpenHAB::DSL::MonkeyPatch::Items::MetadataItem], or nil if the namespace doesn't exist
+            #
+            def dig(key, *keys)
+              return self[key.to_s] if keys.empty?
```

Comment:
If we want to support key as symbol, perhaps implement it in the parent NamespaceAccessor::[] and remove key.to_s from here. However, to be consistent, we'd need to support it all the way into the configuration, and for that the MetadataItem needs to implement hash with indifference access, something that I didn't tackle in the original implementation.

If you could make the nested configuration hash support symbols too, that'd be great. Otherwise, for the purpose of this PR, perhaps remove the symbol support for now.

Reply:
I am also wondering whether we should return the .value here or return the MetadataItem

Reply:
I could drop the symbol, no problem

Reply:
> I am also wondering whether we should return the .value here or return the MetadataItem

Not sure either, returning the MetadataItem would be more consistent with the [] method.

Reply:
> Not sure either, returning the MetadataItem would be more consistent with the [] method.

If it returns the value, it would make dig always return a value wether it's the namespace's value or the config value. OTOH, it would be inconsistent because Item.meta.dig('namespace') should technically be the same as Item.meta['namespace'], right?

@boc-tothefuture WDYT?

Reply:
The more I think about it, the more I think that dig('namespace') should return the value, not the MetadataItem. If the user wants to get the MetadataItem, they could just use Item.meta['namespace'].

Another reason: consider this metadata definition in an item:
```
Switch Switch_Item { namespace1="ns_value" [ config1="conf1_value", config2="conf2_value" ] }
```

It seems clear to me that dig('namespace1') should return 'ns_value', whereas dig('namespace1', 'config1') should return 'conf1_value'.

In any case I don't think it matters that much because the main reason for using dig would be to get into the configuration values. However, we do need to make a decision here.

So my vote, is for dig('namespace') to return the value, i.e.
```ruby
            def dig(key, *keys)
              return self[key]&.value if keys.empty?
```

Reply:
I can agree with that, will change so it returns the value